modules/data.land/R/shp2gpkg.R

```diff
@@ -0,0 +1,79 @@
+#' Convert Shapefile to GeoPackage
+#'
+#' This function converts a Shapefile to a GeoPackage using the `sf` package.
+#'
+#' @param input_shp Character. Path to the input Shapefile (e.g., `"data/myfile.shp"`).
+#' @param output_gpkg Character. Path to the output GeoPackage (e.g., `"output/myfile.gpkg"`).
+#' @param layer_name Character. Optional. Name of the layer in the GeoPackage. Defaults to the base name of the Shapefile.
+#' @param overwrite Logical. Whether to overwrite an existing layer in the GeoPackage. Defaults to `TRUE`.
+#'
+#' @return Invisibly returns the path to the GeoPackage (`output_gpkg`) upon successful conversion.
+#' @details
+#' This function reads a Shapefile, converts it to an `sf` object, and writes it to a GeoPackage.
+#' The `sf` package handles the conversion and ensures spatial data integrity.
+#'
+#' @examples
+#' # Convert 'roads.shp' to 'roads.gpkg' with the default layer name
+#' shp2gpkg("data/roads.shp", "output/roads.gpkg")
+#'
+#' # Convert with a custom layer name
+#' shp2gpkg("data/roads.shp", "output/roads.gpkg", layer_name = "custom_layer")
+#'
+#' # Prevent overwriting existing layers
+#' shp2gpkg("data/roads.shp", "output/roads.gpkg", overwrite = FALSE)
+#'
+#' @importFrom sf st_read st_write
+#' @export
+shp2gpkg <- function(input_shp, output_gpkg, layer_name = NULL, overwrite = TRUE) {
+  # Load the Shapefile
+  shapefile <- sf::st_read(input_shp, quiet = TRUE)
+  
+  # Check validity of geometries
```

Comment:
@camalmborg this is the logic for reparing geometries. I could move it to a separate function if you think it will be useful